jerry-core/vm/vm-stack.c

```diff
@@ -323,62 +368,71 @@ vm_stack_find_finally (vm_frame_ctx_t *frame_ctx_p, /**< frame context */
         exception = jcontext_take_exception ();
       }
 
-      ecma_value_t iterator = stack_top_p[-3];
-      ecma_value_t result = ecma_op_get_method_by_magic_id (iterator, LIT_MAGIC_STRING_RETURN);
+      ecma_value_t result;
 
-      if (!ECMA_IS_VALUE_ERROR (result) && !ecma_is_value_undefined (result))
+      if (context_type == VM_CONTEXT_ITERATOR)
       {
-        if (!ecma_is_value_object (result) || !ecma_op_is_callable (result))
-        {
-          ecma_free_value (result);
-          result = ecma_raise_type_error (ECMA_ERR_MSG ("Iterator 'return' is not callable"));
-        }
-        else
-        {
-          ecma_object_t *return_obj_p = ecma_get_object_from_value (result);
-          result = ecma_op_function_call (return_obj_p, iterator, NULL, 0);
-          ecma_deref_object (return_obj_p);
+        result = ecma_op_iterator_close (stack_top_p[-2]);
+      }
+      else
+      {
+        ecma_value_t iterator = stack_top_p[-3];
+        result = ecma_op_get_method_by_magic_id (iterator, LIT_MAGIC_STRING_RETURN);
 
-          if (context_type == VM_CONTEXT_FOR_AWAIT_OF && !ECMA_IS_VALUE_ERROR (result))
+        if (!ECMA_IS_VALUE_ERROR (result) && !ecma_is_value_undefined (result))
+        {
+          if (!ecma_is_value_object (result) || !ecma_op_is_callable (result))
```

Comment:
The `!ecma_op_is_callable (result)` check should be enough. However if I read this correctly this is just indentation change so maybe we'll change this in a different PR.